### PR TITLE
Adds _PYPY to twisted.python.compat

### PR DESCRIPTION
**Pull request by rodrigc**
Taken from https://github.com/twisted/twisted/pull/221

See: https://twistedmatrix.com/trac/ticket/8503

### CODE REVIEW

**Comment by rodrigc** [comment]
@markrwilliams I am trying to merge your patch, but the codecov checks which were recently made mandatory are preventing me from doing the merge.  See: https://github.com/twisted/twisted/pull/223 , in the section **Some checks were not successful**

Your changes are legit, so there is nothing on your side.

@adiroiban @glyph I think we need to add pypy to the list of coverage reports that are uploaded to codecov.  I don't know how to do that.  Can you help?

File: twisted/persisted/styles.py [hunk 1]
```diff
@@ -262,8 +261,8 @@ def __reduce__(self):
 
     def __getstate__(self):
         log.msg( "WARNING: serializing ephemeral %s" % self )
-        import gc
-        if '__pypy__' not in sys.builtin_module_names:
+        if _PYPY:
```
Comment (alex) [inline twisted/persisted/styles.py]: pretty sure this condition is backwards, should be `if not _PYPY`

Comment (markrwilliams) [inline twisted/persisted/styles.py]: you are right :(

**Comment by adiroiban** [comment]
this is ugly, but yes, we need to add a coverage report for pypy.

I am trying to get coverage report for pypi....see https://github.com/twisted-infra/braid/issues/209

I would say that if someone cares about Twisted on PyPy, the work should start by fixing the current tests to make PyPy a supported platform.

**Comment by rodrigc** [comment]
@adiroiban At the Twisted Meetup in San Francisco, @markrwilliams mentioned that he was interested in fixing the tests that are breaking on pypy, so he is working on it.

**Comment by markrwilliams** [comment]
That's exactly what I'm aiming to do :)  I've got set of changes that (seem
to) make all tests pass on PyPy. I'm trying to break them up into
individual patches/PRs to make review easier.

If it makes getting coverage working easier I can push them as one
patch/PR.  The diff is +150/-35 so it's not huge.

**Comment by rodrigc** [comment]
@markrwilliams hold on and wait for the pypy coverage buildbot to be set up.
Splitting up into small patches is a good approach.  That is what I have been doing for Python3 stuff, and it has worked out well.

**Comment by markrwilliams** [comment]
@rodrigc you got it :)

FWIW I think addressing @alex's comment will restore coverage and hopefully solve the mergability problem.

**Comment by adiroiban** [comment]
I am trying to build pypy with travis-ci ... and have it in the allow_failures

**Comment by markrwilliams** [comment]
@adiroiban ok!  thank you!!

**Comment by adiroiban** [comment]
The PyPy Travis-CI part has this ticket https://twistedmatrix.com/trac/ticket/8505 ... hope it will be easy :)

**Comment by adiroiban** [comment]
See https://github.com/twisted/twisted/pull/226 ... I am stuck as it looks like coverage is not working on PyPy. 

@markrwilliams please take a look and see if you can help. Thanks!

I am a simple py2.7 gui and I have never used the fancy py3 or pypy versions :)

**Comment by markrwilliams** [comment]
@adiroiban good news - i can reproduce the coverage failure on my laptop.  i'll dig in and see if i can figure things out.  thanks!!

**Comment by adiroiban** [comment]
@markrwilliams thanks! Please send your feedback to https://github.com/twisted/twisted/pull/226 so that we can have it fixed in trunk.

**Comment by rodrigc** [comment]
@glyph @adiroiban Can you override codecov and merge this?  codecov is blocking this,
but only because the PyPy builder which can upload coverage results to codecov is not set up yet.
This is blocking @markrilliams from moving forward with PyPy work.

**Comment by rodrigc** [comment]
Closing this due to this review feedback: https://twistedmatrix.com/trac/ticket/8503#comment:7

**Comment by rodrigc** [comment]
@glyph based on your latest feedback, can you merge this PR?  codecov is blocking it.
This PR will help lay the foundation to get a working Pypy buildbot

**Comment by rodrigc** [comment]
I would have preferred this to get merged instead of https://github.com/twisted/twisted/pull/221 .
But https://github.com/twisted/twisted/pull/221 was merged, so closing this.

**Pull request closed without merging**